### PR TITLE
Make border radii clip their contents.

### PR DESCRIPTION
**Pull request by pcwalton**
Needed for browser.html rounded corners.

r? @mbrubeck

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9810)

### CODE REVIEW

**Comment by paulrouget** [comment]
Does this fix https://github.com/servo/servo/issues/7575 ?

File: tests/wpt/mozilla/tests/css/border_radius_clipping_contents_a.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<style>
```
Comment (jdm) [inline tests/wpt/mozilla/tests/css/border_radius_clipping_contents_a.html]: Missing the appropriate `<link>` for the reference file.

File: tests/wpt/mozilla/meta/MANIFEST.json [hunk 2]
```diff
@@ -776,6 +776,18 @@
...
+            "path": "css/border_radius_clipping_contents_a.html",
+            "references": [
+              [
+                "/_mozilla/css/border_radius_clipping_contents_ref.html",
+                "=="
+              ]
+            ],
+            "url": "/_mozilla/css/border_radius_clipping_contents_a.html"
+          }
+        ],
```
Comment (jdm) [inline tests/wpt/mozilla/meta/MANIFEST.json]: This is missing a second entry later on in the file. It's much easier to use `./mach create-wpt --reftest path/to/test.html --reference path/to/reference.html` which does everything right the first time.

**Comment by paulrouget** [comment]
I think this is breaking this example:

```html
<div></div>

<style>
  div {
  display: inline-block;
  width: 20px;
  height: 20px;
  margin-left: 20px;
  background: red;
  border-radius: 50%;
}
</style>
```

**Comment by jdm** [comment]
@pcwalton Did the most recent commit address @paul's earlier testcase?

**Comment by pcwalton** [comment]
Not yet. Looking into it now.

**Comment by pcwalton** [comment]
I tested that example and it seems to work now. Also updated the reftest.

r? @mbrubeck

**Comment by mbrubeck** [comment]
@bors-servo r+

---

Reviewed 1 of 4 files at r1, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9810#-:-KCXTS-WmQPKg4MyDPcN:1236647940)*

**Comment by jdm** [comment]
```
  ▶ FAIL [expected PASS] /_mozilla/css/border_radius_clipping_contents_a.html
  └   → /_mozilla/css/border_radius_clipping_contents_a.html 9eb7aecdd5ea65f459f27c3528048aaedefb9fb8
/_mozilla/css/border_radius_clipping_contents_ref.html 422b9bf4ebb144b6454657178ddc1139d0ae8570
Testing 9eb7aecdd5ea65f459f27c3528048aaedefb9fb8 == 422b9bf4ebb144b6454657178ddc1139d0ae8570
```

**Comment by mbrubeck** [comment]
Looks like an antialising problem; the antialiased pixels are darkened instead of lightened.

screenshot at 2016-03-11 10 22 45

**Comment by pcwalton** [comment]
Yeah, the problem is that the red shows through slightly due to the antialiasing. I think this is not really a bug in Servo or WebRender; it's how antialiasing is supposed to work when implemented via alpha blending. I just removed the `background: red`.

r? @mbrubeck

**Comment by mbrubeck** [comment]
@bors-servo r+

---

Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9810#-:-KCrvdCsoJ50CXWzdlKM:1236647940)*

**Pull request merged**